pagefaultgames/pokerogue
# [UI/UX][QoL] Ask confirmation before disabling touchpad controls in settings

## What are the changes the user will see?
- Touch-related settings do not get displayed anymore when using a device without touchscreen
- People playing on mobile are no longer prevented from disabling touch controls
- When moving the cursor to the option to disable touch controls, a confirmation prompt is shown before the change is applied


## Why am I making these changes?
- [Some people](https://discord.com/channels/1125469663833370665/1307709907595690025) who play on mobile don't use touch controls and don't want them obscuring the screen, but currently there is a check that prevents disabling controls on mobile
- Said check (touch-controls.isMobile) actually does not work for [all devices](https://discord.com/channels/1125469663833370665/1279236557546586204), so people sometimes lock themselves out of being able to play the game on accident




## What are the changes from a developer perspective?
- Remove the check that prevents disabling touch control on mobile
- Move the two touch related settings next to each other in the settings, and hide them if the device doesn't have a touchscreen
- Make `AbstractSettingsUiHandler` extend `MessageUiHandler` so that they can display text in a dedicated window
- Add new optional attributes `needConfirmation` and `confirmationMessage` in `SettingOption`s that are used to display a confirmation message before changing the setting
- Before saving a setting with the `needConfirmation` attribute, show a confirmation message and Yes/No prompt



## Screenshots/Videos

![image](https://github.com/user-attachments/assets/e0656c74-c44a-4570-9896-7204a92c08bc)

actual video soon TM


## How to test the changes?
[Find the locale PR here](https://github.com/pagefaultgames/pokerogue-locales/pull/64)

`npm run start:dev -- --host` to be able to open the game on any device on your network
**If testing on a mobile device** make sure to go to a private window if you don't have a controller or something else connected or you'll need to clear the local storage to re-enable touch controls after disabling them

On PC, use your browser's built-in function to simulate having a touchscreen and play around with disabling and reenabling the option



## Checklist
- [ ] There is no overlap with another PR? slight overlap with https://github.com/pagefaultgames/pokerogue/pull/3638 since they also make the SettingsUiHandler extend MessageUiHandler

Are there any localization additions or changes? If so:
  - [X] If so, please leave a link to it here: https://github.com/pagefaultgames/pokerogue-locales/pull/64
